controllers/controllers.js

```diff
@@ -15,4 +15,19 @@ function getEndpoints(req, res, next) {
   res.status(200).send({ endpoints });
 }
 
-module.exports = { healthCheck, getAllTopics, getEndpoints };
+function getArticleById(req, res, next) {
+  const { article_id } = req.params;
+  if (Number.isNaN(article_id - 0)) {
+    next({ status: 400, msg: "bad request" });
```

Comment:
This error handling works, but you would need to do it multiple times in different endpoints.
Instead, we can have PSQL do the hard work and catch the errors it throws when we give it the incorrect data type, which will have a specific "code"
Check out the second error handling middleware function in this set of notes for an example https://notes.northcoders.com/courses/js-back-end/error-handling#4-refactoring-to-individual-error-handling-blocks